nodejs/eks/cluster.ts

```diff
@@ -140,6 +184,26 @@ export function createCore(name: string, args: ClusterOptions, parent: pulumi.Co
       }, { parent: parent })).role;
     }
 
+    // Create the EKS cluster IAM Role Provider
+    let creationRole: pulumi.Output<aws.iam.Role>;
+    if (args.creationRole !== undefined) {
+        creationRole = pulumi.output(args.creationRole);
+    } else {
+        creationRole = createEntityRole(name, parent);
+    }
+    const eksCreationRoleArn = creationRole.apply(role => role.arn);
+    const eksClusterProvider = new aws.Provider(`${name}-eksClusterCreatorEntity`, {
+        assumeRole: {
+            roleArn: eksCreationRoleArn.apply(async(arn) => {
+                // wait to assume the IAM Role https://github.com/pulumi/pulumi-aws/issues/673
+                if (!pulumi.runtime.isDryRun()) {
+                    await new Promise(resolve => setTimeout(resolve, 15 * 1000));
```

Comment:
I'd consider bumping this up to 30sec given that we've seen IAM timeout [issues](https://github.com/pulumi/pulumi-eks/issues/204) before.